src/realm/array.hpp

```diff
@@ -427,15 +427,15 @@ class Array : public Node, public ArrayParent {
     {
         QueryStateFindFirst state;
         Finder finder = m_vtable->finder[cond::condition];
-        (this->*finder)(value, start, end, 0, &state);
+        finder(*this, value, start, end, 0, &state);
         return state.m_state;
     }
 
     template <class cond>
     bool find(int64_t value, size_t start, size_t end, size_t baseIndex, QueryStateBase* state) const
     {
         Finder finder = m_vtable->finder[cond::condition];
-        return (this->*finder)(value, start, end, baseIndex, state);
+        return finder(*this, value, start, end, baseIndex, state);
```

Comment:
Ok here, the right finder is called. I am missing the setting of the function ptr.